src/python/WMCore/MicroService/MSPileup/MSPileupData.py

```diff
@@ -253,20 +287,8 @@ def updatePileup(self, doc, rseList=None, validate=True, userDN=''):
             self.logger.error(err)
             return [err.error()]
 
-        # check previous fraction value and add new transition record if it is
-        # differ from current one or does not exist, see logic:
-        # https://gist.github.com/amaltaro/b4f9bafc0b58c10092a0735c635538b5
-        prevTranRecord = transition[-1]
-        if prevTranRecord['containerFraction'] != fraction:
-            customName = customDID(prevTranRecord['customDID'])
-            # preserve previous container fraction
-            transitionRecord = {'containerFraction': prevTranRecord['containerFraction'],
-                                'customDID': customName,
-                                'updateTime': gmtimeSeconds(),
-                                'DN': userDN}
-            transition.append(transitionRecord)
-            doc['transition'] = transition
-            self.logger.info("Added transition record for pileup %s", pname)
+        # update transition record if necessary
+        updateTransitionRecord(doc, userDN, self.logger)
```

Comment:
Given that you pass the logger object, I would suggest to convert `updateTransitionRecord` to a class method instead. I think it is more clear them making it a function.

Reply:
I disagree, functions are easier to tests (that's what I did), and except logger (which is literally print) nothing in this function needs class data. I rather prefer to keep it as function.

Reply:
That's okay. I agree that it's much easier to test a function, in general. However, a class with implementing and consolidating the required logic for its data looks tidy to me and more readable. No need to make any changes in here then.